### ISSUE TITLE
Cannot run: TypeError: Cannot read property 'name' of null

### ISSUE THREAD

**Issue by egeriis** [Khan/flow-to-ts#129]
I'm seeing the same error as described in #127. They hinted at the version published to npm being broken; if that's the case a fix should be tracked and published. If anyone can elaborate I'm happy to help.

```
$ flow-to-ts --version
0.1.4
$ flow-to-ts clientside/packages/components/Typography/createMixin.js
error processing clientside/packages/components/Typography/createMixin.js
TypeError: Cannot read property 'name' of null
    at Object.exit (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/src/transform.js:431:18)
    at NodePath._call (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/path/context.js:53:20)
    at NodePath.call (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/path/context.js:40:17)
    at NodePath.visit (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/path/context.js:97:8)
    at TraversalContext.visitQueue (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/context.js:118:16)
    at TraversalContext.visitMultiple (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/context.js:85:17)
    at TraversalContext.visit (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/context.js:144:19)
    at Function.traverse.node (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/index.js:94:17)
    at NodePath.visit (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/path/context.js:95:18)
    at TraversalContext.visitQueue (/Users/ronni/.config/yarn/global/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/context.js:118:16)
```

**Comment by k4b7** [comment]
@egeriis I ran into a similar error in code that was using an explicit inexact object type.  Can you check to see if your code is using the `...` syntax in any of your object types?

**Comment by egeriis** [comment]
@kevinbarabash We're def. not using that, because we're on an old version of Flow that doesn't support it 😬

**Comment by k4b7** [comment]
I wonder what's causing it. 🤔Can you try to create a minimal reproduction in the playground?

**Comment by joebowbeer** [comment]
I suggest checking the issues list. Many of the reported issues manifest as this error message.

Examples: #134 #131

**Comment by initFabian** [comment]
I believe v0.1.6 broke functionality some functionality around union types. I had to drop my version to v0.1.5 to get the following code to work.

```
// @flow

type AcceptableDateFormat = 'default-date-format' | 'date-only-format' | 'time-only-format' | 'time-date-format' | 'date-time-full-format'

const DATE_ONLY = 'MM/DD/YY';
const TIME_ONLY = 'h:mm A';
const TIME_DATE = 'h:mm A MM/DD/YY UTC';
const DATE_TIME_FULL = 'MM/DD/YY, hh:mm:ss.SSS A UTC';
export const DEFAULT_DATE_FORMAT = 'MM/DD/YY, hh:mm A UTC';

type DateFormatKeyDictionaryType = {
  [x: AcceptableDateFormat]: string
}

const DateFormatKeyDictionary: DateFormatKeyDictionaryType = {
  'default-date-format': DEFAULT_DATE_FORMAT,
  'date-only-format': DATE_ONLY,
  'time-only-format': TIME_ONLY,
  'time-date-format': TIME_DATE,
  'date-time-full-format': DATE_TIME_FULL
};
```

**Comment by k4b7** [comment]
@initFabian I'm not seeing any errors when I tried this on https://flow-to-ts.netlify.com/.  Could you provide more details about how 0.1.6 isn't working with the code you posted?

**Comment by initFabian** [comment]
@kevinbarabash - Yes, It does work when using https://flow-to-ts.netlify.com/, but when running 0.1.6 locally it spits back the `TypeError: Property name expected type of string but got null`.

```
$ node_modules/.bin/flow-to-ts -V
0.1.6
$ node_modules/.bin/flow-to-ts ./flow-example.js 
error processing ./flow-example.js
TypeError: Property name expected type of string but got null
    at Object.validate (/Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@babel/types/lib/definitions/utils.js:164:13)
    at validateField (/Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@babel/types/lib/validators/validate.js:22:9)
    at validate (/Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@babel/types/lib/validators/validate.js:16:3)
    at builder (/Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@babel/types/lib/builders/builder.js:38:27)
    at Object.TSTypeParameter (/Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@babel/types/lib/builders/generated/index.js:1182:31)
    at /Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@khanacademy/flow-to-ts/src/transform.js:564:35
    at Array.forEach (<anonymous>)
    at Object.exit (/Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@khanacademy/flow-to-ts/src/transform.js:554:16)
    at NodePath._call (/Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/path/context.js:55:20)
    at NodePath.call (/Users/fabianbuentello/Desktop/flow-to-typescript-codebase/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/path/context.js:42:17)
```

**Comment by k4b7** [comment]
@initFabian thanks for confirming.  That's unfortunate that there are differences between the command line behavior and the playground behavior.  I'll look into it.

**Comment by FezVrasta** [comment]
I can confirm the below code compiles on the playground, but fails locally:

```js
// @flow
type T = {
    [number | string]: number;
};
```

**Comment by masylum** [comment]
I'm facing the same exact issue with:

```js
type Segment = 'hrs' | 'min'
type Range = { [Segment]: { start: number, end: number } }
```

so it looks like a combination of unions + hashes as maps.

**Comment by kigh-ota** [comment]
I report what I have checked concerning this issue. I'm using `@khanacademy/flow-to-ts@0.1.6`.

A simpler code can reproduce the issue:

```js
// @flow
type Entries = { [Key]: string };
type Key = string;
```

```console
$ node_modules/.bin/flow-to-ts test.js
error processing test.js
TypeError: Property name expected type of string but got null
    at Object.validate (/Users/k000552/tmp/node_modules/@babel/types/lib/definitions/utils.js:160:13)
    at validateField (/Users/k000552/tmp/node_modules/@babel/types/lib/validators/validate.js:24:9)
    at validate (/Users/k000552/tmp/node_modules/@babel/types/lib/validators/validate.js:17:3)
    at builder (/Users/k000552/tmp/node_modules/@babel/types/lib/builders/builder.js:38:27)
    at Object.TSTypeParameter (/Users/k000552/tmp/node_modules/@babel/types/lib/builders/generated/index.js:1197:31)
    at indexers.forEach.indexer (/Users/k000552/tmp/node_modules/@khanacademy/flow-to-ts/src/transform.js:564:35)
    at Array.forEach (<anonymous>)
    at Object.exit (/Users/k000552/tmp/node_modules/@khanacademy/flow-to-ts/src/transform.js:554:16)
    at NodePath._call (/Users/k000552/tmp/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/path/context.js:55:20)
    at NodePath.call (/Users/k000552/tmp/node_modules/@khanacademy/flow-to-ts/babel-traverse/lib/path/context.js:42:17)
```

And with a slight change I can avoid the error:

```js
// @flow
type Entries = { [string]: string };
```

```console
$ node_modules/.bin/flow-to-ts test2.js

type Entries = {
  [key: string]: string;
};
```

So it seems that using a user-defined type as a type of object properties is sufficient to cause the issue.
As a workaround, you can probably replace user-defined types with `string` or `number` from all object property types.

**Comment by k4b7** [comment]
I've published [v0.2.1](https://github.com/Khan/flow-to-ts/releases/tag/v0.2.1) which should solve this.  Please let me know if you're still experiencing issues with this.

**Comment by FezVrasta** [comment]
Thanks so much for the fix!! It's working perfectly now. 🚀
